src/components/Form/Dropdown.js

```diff
@@ -2,31 +2,28 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { ButtonGroup, DropdownButton, MenuItem, noop } from 'patternfly-react';
 
-export const Dropdown = ({ id, value, disabled, onChange, onBlur, choices }) => (
-  <ButtonGroup justified key={id}>
-    <DropdownButton
-      id={id}
-      bsStyle="default"
-      className="kubevirt-dropdown"
-      title={value}
-      onSelect={onChange}
-      onBlur={onBlur}
-      disabled={disabled}
-    >
-      {choices.map(choice => {
-        const isObject = typeof choice === 'object';
-        const key = isObject ? choice.id || choice.name : choice;
-        const val = isObject ? choice.name : choice;
+export const Dropdown = ({ id, value, disabled, onChange, onBlur, choices }) => {
+  const title = typeof value === 'object' ? value.name || value.id : value;
+  return (
+    <ButtonGroup justified key={id}>
+      <DropdownButton id={id} bsStyle="default" className="kubevirt-dropdown" title={title} disabled={disabled}>
+        {choices.map(choice => {
+          const isObject = typeof choice === 'object';
+          const key = isObject ? choice.id || choice.name : choice;
+          const val = isObject ? choice.name : choice;
+          const onSelect = isObject ? () => onChange(key, val) : () => onChange(key);
+          const blur = isObject ? () => onBlur(key, val) : () => onBlur(key);
```

Comment:
The blur is not much useful here. I guess we could remove the callback altogether.